pkg/resource/plugin/analyzer_config.go

```diff
@@ -0,0 +1,280 @@
+			return nil, err
+		}
+
+		el, err := extractEnforcementLevel(props)
+		if err != nil {
+			return nil, errors.Wrapf(err, "parsing enforcement level for %q", k)
+		}
+		enforcementLevel = el
+		if len(props) > 0 {
+			properties = props
+		}
+
+		// Don't bother including empty configs.
+		if enforcementLevel == "" && len(properties) == 0 {
+			continue
+		}
+
+		result[k] = AnalyzerPolicyConfig{
+			EnforcementLevel: enforcementLevel,
+			Properties:       properties,
+		}
+	}
+	return result, nil
+}
+
+func parsePolicyPackConfig(b []byte) (map[string]AnalyzerPolicyConfig, error) {
+	result := make(map[string]AnalyzerPolicyConfig)
+
+	// Gracefully allow empty content.
+	if strings.TrimSpace(string(b)) == "" {
+		return nil, nil
+	}
+
+	config := make(map[string]interface{})
+	if err := json.Unmarshal(b, &config); err != nil {
+		return nil, err
+	}
+	for k, v := range config {
+		var enforcementLevel apitype.EnforcementLevel
+		var properties map[string]interface{}
+		switch val := v.(type) {
+		case string:
+			el := apitype.EnforcementLevel(val)
+			if !el.IsValid() {
+				return nil, errors.Errorf(
+					"parsing enforcement level for %q: %q is not a valid enforcement level", k, val)
+			}
+			enforcementLevel = el
+		case map[string]interface{}:
+			el, err := extractEnforcementLevel(val)
+			if err != nil {
+				return nil, errors.Wrapf(err, "parsing enforcement level for %q", k)
+			}
+			enforcementLevel = el
+			if len(val) > 0 {
+				properties = val
+			}
+		default:
+			return nil, errors.Errorf("parsing %q: %v is not a valid value; must be a string or object", k, v)
+		}
+
+		// Don't bother including empty configs.
+		if enforcementLevel == "" && len(properties) == 0 {
+			continue
+		}
+
+		result[k] = AnalyzerPolicyConfig{
+			EnforcementLevel: enforcementLevel,
+			Properties:       properties,
+		}
+	}
+	return result, nil
+}
+
+// extractEnforcementLevel looks for "enforcementLevel" in the map, and if so, validates that it is a valid value, and
+// if so, deletes it from the map and returns it.
+func extractEnforcementLevel(props map[string]interface{}) (apitype.EnforcementLevel, error) {
+	contract.Assertf(props != nil, "props != nil")
+
+	var enforcementLevel apitype.EnforcementLevel
+	if unknown, ok := props["enforcementLevel"]; ok {
+		enforcementLevelStr, isStr := unknown.(string)
+		if !isStr {
+			return "", errors.Errorf("%v is not a valid enforcement level; must be a string", unknown)
+		}
+		el := apitype.EnforcementLevel(enforcementLevelStr)
+		if !el.IsValid() {
+			return "", errors.Errorf("%q is not a valid enforcement level", enforcementLevelStr)
+		}
+		enforcementLevel = el
+		// Remove enforcementLevel from the map.
+		delete(props, "enforcementLevel")
+	}
+	return enforcementLevel, nil
+}
+
+// ValidatePolicyPackConfig validates the policy pack's configuration.
+func ValidatePolicyPackConfig(policies []AnalyzerPolicyInfo, config map[string]AnalyzerPolicyConfig) error {
```

Comment:
Is this called?

Reply:
Yes, but inside the package. Changed to lowercase.